src/components/Clock.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  currentTime: Date;
+};
+
+export class Clock extends React.PureComponent<Props, State> {
+  state: State = {
+    currentTime: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ currentTime: new Date() });
+      // eslint-disable-next-line no-console
+      console.info(this.state.currentTime.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {this.props.name}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {this.state.currentTime.toUTCString().slice(-12, -4)}
```

Comment:
It is better to put such calculations in separate variables to improve readability